src/kOS/Module/kOSVesselModule.cs

```diff
@@ -223,16 +226,55 @@ private void HookEvents()
         /// </summary>
         private void UnHookEvents()
         {
-            if (RemoteTechHook.IsAvailable(parentVessel.id))
+            if (hasRemoteTech)
             {
-                RemoteTechHook.Instance.RemoveSanctionedPilot(parentVessel.id, UpdateAutopilot);
+                UnHookRemoteTechPilot();
             }
             else
             {
-                parentVessel.OnPreAutopilotUpdate -= UpdateAutopilot;
+                UnHookStockPilot();
             }
         }
 
+        private void CheckSwapEvents()
```

Comment:
@hvacengi : After my tests, I'm ready to merge this but I just wanted your feedback first on that \<summary\> comment I have above - is what it says true?  I'd like to add it to the method if it's truthful.